### PR TITLE
poppler: fix pkg_config generator for consumers + install unstable API + bump dependencies

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **poppler/20.09.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

got errors while trying to add `poppler` support in `gdal` recipe (relies on `pkg_config` generator):

```
gdal/3.2.0: ERROR: Generator pkg_config(file:None) failed
sequence item 0: expected str instance, list found
ERROR: sequence item 0: expected str instance, list found
```

Moreover, gdal relies on poppler unstable API (https://trac.osgeo.org/gdal/ticket/7034)... I've enabled installation of those headers.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]

**Comment by SpaceIm** [comment]
@madebr, what do you think to disable `ENABLE_RELOCATABLE` if static? There are conflicts on `DllMain` between gdal and poppler if gdal is shared.

**Comment by madebr** [comment]
`DllMain` should not be linked in when poppler is built as a static lib.
It would be #ifdef'ed out

**Comment by SpaceIm** [comment]
So it would be fine to remove all this block if static: https://github.com/freedesktop/poppler/blob/73105150eb36d295d5574564421638ce34277fd5/poppler/GlobalParams.cc#L101 ?

**Comment by madebr** [comment]
DllMain is the dll entry point.
It has no point when used as a static library.
It is even causing duplicate symbol errors.

I'm not on a pc right now, is `ENABLE_RELOCATIONS` used elsewhere?

**Comment by SpaceIm** [comment]
No it's not used elsewhere.

File: recipes/poppler/all/conanfile.py [hunk 1]
```diff
@@ -189,7 +189,8 @@ def _configure_cmake(self):
         self._cmake.definitions["ENABLE_DCTDECODER"] = self._dct_decoder
         self._cmake.definitions["USE_FLOAT"] = self.options.float
         self._cmake.definitions["RUN_GPERF_IF_PRESENT"] = False
-        self._cmake.definitions["ENABLE_RELOCATABLE"] = True
+        if self.settings.os == "Windows":
+            self._cmake.definitions["ENABLE_RELOCATABLE"] = self.options.shared
```
Comment (ericLemanissier) [inline recipes/poppler/all/conanfile.py:193]: Isn't this a little risky ? a static library can be linked into a shared library

Comment (madebr) [reply recipes/poppler/all/conanfile.py:193]: This remains possible, the new *owner* must provide its own `DllMain` entry point and make sure that poppler is initialized correctly.
In other words the same as when poppler is used in an ordinary executable.

**Review by madebr — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**